decision-records/2025-04-16--redaction-processor-for-otelcol.md

```diff
@@ -0,0 +1,160 @@
+
+ - Remove any attributes not included in a predefined list of permitted attributes.
+ - Strip confidential information from telemetry to prevent accidental data exposure.
+ - Mask or obfuscate sensitive attribute values that match standard patterns.
+
+
+
+### Config examples
+
+#### Log record in plain text format
+
+In a log line like this one:
+
+```
+2025-04-14T10:45:32Z INFO User john.doe@example.com logged in with token abcdef123456
+```
+
+We need to::
+
+ - Mask the email → `***`
+ - Mask the token → `***`
+
+The `redaction` processor config block looks like:
+
+```yaml
+processors:
+  redaction/redact_email_token:
+    rules:
+      - name: redact_email
+        pattern: '[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}'
+        replacement: '***'
+      - name: redact_token
+        pattern: 'token\s+\w+'
+        replacement: 'token ***'
+```
+
+
+#### Log record in JSON format
+
+```json
+{"timestamp": "2025-04-14T10:45:32Z", "level": "INFO", "user": "john.doe@example.com", "token": "abcdef123456", "action": "login"}
+```
+
+We need to::
+
+ - Mask the email → `***`
+ - Mask the token → `***`
+
+
+The `redaction` processor config block looks like:
+
+```yaml
+processors:
+  redaction/redact_email_token:
+    rules:
+      - name: redact_email
+        pattern: '"user":\s*"[^"]+@[^"]+"'
+        replacement: '"user": "***"'
+      - name: redact_token
+        pattern: '"token":\s*"[^"]+"'
+        replacement: '"token": "***"'
+```
+
+
+#### Use these configs in a `pipeline`
+
+In order to use one of these configs for the `redaction` procesor in a pipeline, we need to add it like this:
+
+```yaml
+service:
+  pipelines:
+    logs/0:
+      receivers:
+      - loki
+      processors:
+      - redaction/redact_email_token
+      exporters:
+      - loki/0
+```
+
+### How Loki stores the redacted logs:
+
+Loki stores the modified logs these ways
+
+#### Log record in plain text format:
+```json
+{
+  "stream": {
+    "job": "my-job",
+    "filename": "/var/log/myapp.log",
+    "level": "INFO"
+  },
+  "values": [
+    [
+      "1713081932000000000",
+      "2025-04-14T10:45:32Z INFO User *** logged in with token ***"
+    ]
+  ]
+}
+```
+
+#### Log record in JSON format:
+
+```json
+{
+  "stream": {
+    "job": "my-job",
+    "filename": "/var/log/myapp.log",
+    "level": "INFO"
+  },
+  "values": [
+    [
+      "1713081932000000000",
+      "{\"timestamp\": \"2025-04-14T10:45:32Z\", \"level\": \"INFO\", \"user\": \"***\", \"token\": \"***\", \"action\": \"login\"}"
+    ]
+  ]
+}
+```
+
+## Interaction through `juju` command line interface
+
+The `redaction` processor confis should be defined in a `yaml` file and their mapping to pipelines like this:
```

Comment:
We should explicitly specify if we expect the yaml content to e.g. include `processors:`. The description must not be ambiguous.